File: LC70.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n 

        a, b, res = 1, 2, 0
        
        for _ in range(2,n):
            res = a + b
            a = b
            b = res
        
        return res

File: LC70.cpp
class Solution {
public:
    int climbStairs(int n) {
        if(n <= 2)
            return n;
        
        int res = 0, a = 1, b = 2;

        for(int i = 2; i < n; i++){
            res = a + b;
            a = b;
            b = res;
        }

        return res;
    }
};
